pkg/flowaggregator/flowaggregator_test.go

```diff
@@ -578,3 +582,225 @@ func TestFlowAggregator_closeUpdateChBeforeFlowExportLoopReturns(t *testing.T) {
+			obj:  pendingPod,
+			want: []string{"192.168.1.2"},
+		},
+		{
+			name: "pod status in succeeded phase",
+			obj:  succeededPod,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := podInfoIndexFunc(tt.obj)
+			if tt.expectedErr != "" {
+				assert.ErrorContains(t, err, tt.expectedErr)
+			} else {
+				require.NoError(t, err)
+				assert.Equal(t, tt.want, got)
+			}
+		})
+	}
+}
+
+func TestFlowAggregator_fetchPodLabels(t *testing.T) {
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "default",
+			Name:      "testPod",
+			Labels: map[string]string{
+				"test": "ut",
+			},
+		},
+		Status: v1.PodStatus{
+			Phase: v1.PodPending,
+			PodIPs: []v1.PodIP{
+				{
+					IP: "192.168.1.2",
+				},
+			},
+		},
+	}
+
+	client := fake.NewSimpleClientset()
+	informerFactory := informers.NewSharedInformerFactory(client, 0)
+	informerFactory.Core().V1().Pods().Informer().AddIndexers(cache.Indexers{podInfoIndex: podInfoIndexFunc})
+
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+
+	informerFactory.Start(stopCh)
+
+	tests := []struct {
+		name       string
+		podAddress string
+		addPod     bool
+		want       string
+	}{
+		{
+			name:       "no pod objects",
+			podAddress: "192.168.1.2",
+		},
+		{
+			name:       "pod with label",
+			podAddress: "192.168.1.2",
+			addPod:     true,
+			want:       "{\"test\":\"ut\"}",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			if tt.addPod {
+				informerFactory.Core().V1().Pods().Informer().GetIndexer().Add(pod)
+			}
+
+			fa := &flowAggregator{
+				k8sClient:        client,
+				includePodLabels: true,
+				podInformer:      informerFactory.Core().V1().Pods(),
+			}
+			got := fa.fetchPodLabels(tt.podAddress)
+			assert.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func TestFlowAggregator_GetRecordMetrics(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	mockCollectingProcess := ipfixtesting.NewMockIPFIXCollectingProcess(ctrl)
+	mockAggregationProcess := ipfixtesting.NewMockIPFIXAggregationProcess(ctrl)
+
+	tests := []struct {
+		name               string
+		collectingProcess  ipfix.IPFIXCollectingProcess
+		aggregationProcess ipfix.IPFIXAggregationProcess
+		numRecordsExported int64
+		want               querier.Metrics
+	}{
+		{
+			name:               "one numRecordsExported",
+			collectingProcess:  mockCollectingProcess,
+			aggregationProcess: mockAggregationProcess,
+			numRecordsExported: 1,
+			want: querier.Metrics{
+				NumRecordsExported: 1,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fa := &flowAggregator{
+				collectingProcess:  tt.collectingProcess,
+				aggregationProcess: tt.aggregationProcess,
+				numRecordsExported: tt.numRecordsExported,
+			}
+
+			mockCollectingProcess.EXPECT().GetNumRecordsReceived()
+			mockAggregationProcess.EXPECT().GetNumFlows()
+			mockCollectingProcess.EXPECT().GetNumConnToCollector()
```

Comment:
why not return actual values here and check that they are included in `got`

Reply:
done.